tools/apiview/parsers/cpp-api-parser/ParseAzureSdkCpp/ParseAzureSdkCpp.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// ParseAzureSdkCpp.cpp : Defines the entry point for the application.
+//
+// This file primarily parses the command line (using TCLAP
+// (https://tclap.sourceforge.net/manual.html) for command line parsing).
+//
+// It then feeds those inputs into the ApiViewProcessor and dumps an output JSON file whose name is
+// provided on the command line.
+//
+
+#include "ApiViewProcessor.hpp"
+#include "JsonDumper.hpp"
+#include "TextDumper.hpp"
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <ostream>
+#include <tclap/CmdLine.h>
+#include <vector>
+
+int main(int argc, char** argv)
+{
+  try
+  {
+    TCLAP::CmdLine commandLine("C++ ApiView Parser", ' ');
+
+    TCLAP::UnlabeledValueArg<std::string> inputDirectory(
+        "input", "Input Directory", true, ".", "string", commandLine);
+    TCLAP::ValueArg<std::string> outputFileArg(
+        "o", "output", "Output filename", true, "ApiReview.json", "string", commandLine);
+    TCLAP::ValueArg<std::string> reviewName(
+        "r", "review", "Review Name", false, "", "string", commandLine);
+    TCLAP::ValueArg<std::string> packageVersion(
+        "", "packageVersion", "Package Version", false, "", "string", commandLine);
+    TCLAP::SwitchArg consoleOutput("c", "console", "Dump output to console (diagnostic)", commandLine);
+
+    commandLine.parse(argc, argv);
+    std::filesystem::path outputFileName{std::filesystem::absolute(outputFileArg.getValue())};
+
+    if (outputFileName.extension() != ".json")
+    {
+      std::cerr << "Output file name must have an extension of .json" << std::endl;
+      return 1;
+    }
+
+    std::string directoryToParse{inputDirectory.getValue()};
+    if (directoryToParse.back() == '\\')
+    {
+      directoryToParse.erase(directoryToParse.size() - 1);
+    }
+    ApiViewProcessor apiViewProcessor(directoryToParse);
+
+    int rv = apiViewProcessor.ProcessApiView();
+    if (rv == 0)
```

Comment:
rv once set to 0 will not change if an error occurs in this loop.  Is that your intended behavior?

Reply:
Basically, yes. 

I have an issue I need to deal with at some point that compilation failures from llvm/clang don't cause the tool.run command (which drives the return value from ProcessApiView) to fail - that's a problem because we'll blindly ignore any compilation errors.

But in general, all errors are reported via exceptions, which triggers a -1 return value. Theoretically I could change line 77 to be a "return 0".